src/nnet3/nnet-am-decodable-simple.cc

```diff
@@ -134,17 +135,17 @@ void NnetDecodableBase::EnsureFrameIsComputed(int32 subsampled_frame) {
   int32 first_input_frame = first_output_frame - left_context,
       last_input_frame = last_output_frame + right_context,
       num_input_frames = last_input_frame + 1 - first_input_frame;
-  Vector<BaseFloat> ivector;
-  GetCurrentIvector(first_output_frame,
-                    last_output_frame - first_output_frame,
-                    &ivector);
+  Matrix<BaseFloat> ivectors;
+  GetIvectorsForFrames(first_output_frame,
+                       last_output_frame - first_output_frame,
+                       &ivectors);
```

Comment:
The way you are deciding which ranges of frames you want to get iVectors for is going to severely hurt LSTM performance.  You are only giving it the minimum time-range of iVectors that it needs to compute the output.  But if you give it extra input (as you do in LSTMs with --extra-left-context) you're not giving it iVectors for those times, so those early layers will become non-computable and the left context will be wasted.
The range needs to be worked out from the input, not output frames-- you can round down to the nearest multiple ivector_period to get the exact first and last frame for which you need the iVector.